### ISSUE TITLE
Application freezes when loading OGG audio

### ISSUE THREAD

**Issue by criezy** [adventuregamestudio/ags#1694]
**Describe the bug**
When starting a new game in Gemini Rue (version from GOG), the AGS executable freezes.

**AGS Version**
AGS commit 267874c54cde170d9cba9d7dc2c50e44cd0b906c from master branch (Friday June 17).

**Game**
Gemini Rue (version from GOG: https://www.gog.com/game/gemini_rue)

**To Reproduce**
1. Start Gemini Rue
2. Start a new game
3. Wait a few seconds watching the intro sequence.
4. AGS freezes

**Expected behavior**
The game should start.

**Desktop (please complete the following information):**
 - OS: macOS
 - Version 11.6

**Additional context**
Here is the backtrace for the freeze. It seems to be trying to load OGG audio.
```
  * frame #0: ags`long long const& std::__1::max<long long, std::__1::__less<long long, long long> >(__a=0x000000016fdf8ac0, __b=0x000000016fdf8ae0, __comp=__less<long long, long long> @ 0x000000016fdf8980) at algorithm:2660:24
    frame #1: ags`long long const& std::__1::max<long long>(__a=0x000000016fdf8ac0, __b=0x000000016fdf8ae0) at algorithm:2669:12
    frame #2: ags`AGS::Common::BufferedStream::Seek(this=0x00000001064f1a20, offset=0, origin=kSeekCurrent) at bufferedstream.cpp:191:26
    frame #3: ags`AGS::Engine::SDL2Util::rwops_seek(context=0x000000012c699840, offset=0, whence=1) at sdl2_util.cpp:47:58
    frame #4: ags`SDL_RWtell_REAL(context=0x000000012c699840) at SDL_rwops.c:794:12
    frame #5: ags`SDL_RWtell(a=0x000000012c699840) at SDL_dynapi_procs.h:778:1
    frame #6: ags`stb_vorbis_get_file_offset(f=0x000000012bc24c80) at stb_vorbis.h:4675:27
    frame #7: ags`vorbis_find_page(f=0x000000012bc24c80, end=0x000000016fdf9140, last=0x000000016fdf9150) at stb_vorbis.h:4696:35
    frame #8: ags`stb_vorbis_stream_length_in_samples(f=0x000000012bc24c80) at stb_vorbis.h:5117:15
    frame #9: ags`VORBIS_open(sample=0x000000012bb48590, ext="ogg") at SDL_sound_vorbis.c:125:18
    frame #10: ags`init_sample(funcs=0x00000001014cce00, sample=0x000000012bb48590, ext="ogg", _desired=0x0000000000000000) at SDL_sound.c:353:10
    frame #11: ags`Sound_NewSample(rw=0x000000012c699840, ext="ogg", desired=0x0000000000000000, bSize=65536) at SDL_sound.c:448:29
    frame #12: ags`AGS::Engine::SDLDecoder::Open(this=0x000000012c6998b0, pos_ms=0) at sdldecoder.cpp:98:39
    frame #13: ags`audio_core_slot_init(in=nullptr, extension_hint=0x000000016fdfa0e0, repeat=false) at audio_core.cpp:331:19
    frame #14: ags`my_load_clip(apath=0x000000016fdfa380, extension_hint="OGG", loop=false) at sound.cpp:205:16
    frame #15: ags`my_load_ogg(asset_name=0x000000016fdfa380, loop=false) at sound.cpp:232:12
    frame #16: ags`load_sound_clip(audioClip=0x000000010e815830, repeat=false) at audio.cpp:234:21
    frame #17: ags`load_sound_clip_from_old_style_number(isMusic=true, indexNumber=3, repeat=false) at audio.cpp:547:16
    frame #18: ags`load_music_from_disk(mnum=3, doRepeat=false) at audio.cpp:1063:25
    frame #19: ags`play_new_music(mnum=3, music=0x0000000000000000) at audio.cpp:1112:20
    frame #20: ags`newmusic(mnum=3) at audio.cpp:1130:5
    frame #21: ags`PlayMusicResetQueue(newmus=3) at global_audio.cpp:172:5
    frame #22: ags`Sc_PlayMusicResetQueue(params=0x000000016fdfae10, param_count=1) at global_api.cpp:1210:5
    frame #23: ags`ccInstance::Run(this=0x000000011c94b080, curpc=4275) at cc_instance.cpp:1140:28
    frame #24: ags`ccInstance::CallScriptFunction(this=0x000000011c94b080, funcname="room_AfterFadeIn", numargs=0, params=0x0000000000000000) at cc_instance.cpp:395:18
    frame #25: ags`RunScriptFunction(sci=0x000000011c94b080, tsname="room_AfterFadeIn", numParam=0, params=0x0000000000000000) at script.cpp:405:30
    frame #26: ags`RunScriptFunctionInRoom(tsname="room_AfterFadeIn", param_count=0, params=0x0000000000000000) at script.cpp:444:17
    frame #27: ags`RunScriptFunctionAuto(sc_inst=kScInstRoom, tsname="room_AfterFadeIn", param_count=0, params=0x0000000000000000) at script.cpp:484:16
    frame #28: ags`QueueScriptFunction(sc_inst=kScInstRoom, fn_name="room_AfterFadeIn", param_count=0, params=0x0000000000000000) at script.cpp:321:9
    frame #29: ags`run_interaction_script(nint=0x000000012c758690, evnt=7, chkAny=-1) at script.cpp:221:13
    frame #30: ags`process_event(evp=0x000000010730e674) at event.cpp:204:13
    frame #31: ags`processallevents() at event.cpp:402:9
    frame #32: ags`game_loop_update_events() at game_run.cpp:716:5
    frame #33: ags`UpdateGameOnce(checkControls=true, extraBitmap=0x0000000000000000, extraX=0, extraY=0) at game_run.cpp:840:5
    frame #34: ags`GameTick() at game_run.cpp:991:5
    frame #35: ags`RunGameUntilAborted() at game_run.cpp:1092:9
    frame #36: ags`initialize_start_and_play_game(override_start_room=0, load_save=) at game_start.cpp:129:9
    frame #37: ags`initialize_engine(startup_opts=size=0) at engine.cpp:1311:5
    frame #38: ags`ags_entry_point(argc=2, argv=0x000000016fdff8a0) at main.cpp:423:18
    frame #39: ags`main(argc=2, argv=0x000000016fdff8a0) at main_sdl2.cpp:5:12
```

**Comment by ivan-mogilko** [comment]
@criezy thank you for report, I found that our custom RWOps Seek implementation did not return proper error value.
Pushed a fix: 27c358a04ea0a1f84f60f4e767f43e301a0868ed

**Comment by criezy** [comment]
Thank you. That was quick!
And I confirm that the commit you pushed fixed it for me as well.
